AWSIoTFreeRTOSOTATests/OTATests.swift

```diff
@@ -0,0 +1,315 @@
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import XCTest
+import AWSIoT
+
+class OTATests: XCTestCase {
+
+    /// Expectatioun timeout value for any network operations
+    static let networkTimeout: TimeInterval = 120
+
+    static let dataManagerServiceKey = "OTATests-datamanager"
+
+    /// Membership in this Thing Group will cause the OTA job to be queued for the thing
+    static let thingGroupName = "OTA_INTEG_TEST"
+
+    // MARK: - Thing properties
+
+    let thingName: String = {
+        let timestamp = IoTTestHelpers.thingNameDateFormatter.string(from: Date())
+        return "OTATests-TestThing-\(timestamp)"
+    }()
+
+    // MARK: - Test lifecycle
+
+    override static func setUp() {
+        AWSDDLog.sharedInstance.logLevel = .debug
+        AWSDDLog.add(AWSDDTTYLogger.sharedInstance)
+
+        AWSTestUtility.setupCognitoCredentialsProvider()
+
+        // Default IoT will be used for managing Things and Groups
+        // IoT DataManager used to subscribe to topics must use the ATS-enabled data endpoint
+        let defaultCredentialsProvider = AWSServiceManager.default().defaultServiceConfiguration.credentialsProvider
+        let endpointString = AWSTestUtility.getIoTEndPoint("iot-us-east1-endpoint")!
+        let endpoint = AWSEndpoint(urlString: endpointString)
+
+        // Stream create/describe/list operations do not use the custom endpoint specified by
+        // `AWSTestUtility.getIoTEndPoint("iot-us-east1-endpoint")`. Instead, these operations go to the global service
+        // URL (e.g., "iot.us-east-1.amazonaws.com"
+        let iotConfig = AWSServiceConfiguration(region: .USEast1,
+                                                endpoint: endpoint,
+                                                credentialsProvider: defaultCredentialsProvider)!
+
+        let lwt = AWSIoTMQTTLastWillAndTestament()
+        lwt.message = "OTATest-lwt"
+        lwt.topic = "OTATest-lwt-topic"
+        lwt.qos = .messageDeliveryAttemptedAtLeastOnce
+
+        let mqttConfig = AWSIoTMQTTConfiguration(keepAliveTimeInterval: 30.0,
+                                                 baseReconnectTimeInterval: 1.0,
+                                                 minimumConnectionTimeInterval: 20,
+                                                 maximumReconnectTimeInterval: 8,
+                                                 runLoop: .current,
+                                                 runLoopMode: RunLoop.Mode.default.rawValue,
+                                                 autoResubscribe: true,
+                                                 lastWillAndTestament: lwt)
+
+        AWSIoTDataManager.register(with: iotConfig,
+                                   with: mqttConfig,
```

Comment:
It's not a best practice. In this case, a better name might be something like `register(withIoTConfig iotConfig:mqttConfig:)` to disambiguate. The existing naming is likely an artifact of the way Objective-C APIs are named when they're auto-imported into Swift. We can consider renaming that if it becomes a point of confusion, but in the absence of feedback to that effect, I'd avoid making the change preemptively since it would break existing customers that use this API.